plugins/org.eclipse.fordiac.ide.application/src/org/eclipse/fordiac/ide/application/handlers/MoveThroughHierarchyHandler.java

```diff
@@ -0,0 +1,189 @@
+package org.eclipse.fordiac.ide.application.handlers;
+
+import static org.eclipse.fordiac.ide.model.ui.editors.HandlerHelper.getCommandStack;
+import static org.eclipse.fordiac.ide.model.ui.editors.HandlerHelper.getFBNetwork;
+import static org.eclipse.fordiac.ide.model.ui.editors.HandlerHelper.openEditor;
+
+import java.util.List;
+
+import org.eclipse.core.commands.AbstractHandler;
+import org.eclipse.core.commands.ExecutionEvent;
+import org.eclipse.core.commands.ExecutionException;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.draw2d.geometry.Rectangle;
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.fordiac.ide.application.commands.BorderCrossingReconnectCommand;
+import org.eclipse.fordiac.ide.application.commands.MoveElementsFromSubAppCommand;
+import org.eclipse.fordiac.ide.application.utilities.SubAppHierarchyDialog;
+import org.eclipse.fordiac.ide.model.libraryElement.Application;
+import org.eclipse.fordiac.ide.model.libraryElement.FBNetwork;
+import org.eclipse.fordiac.ide.model.libraryElement.FBNetworkElement;
+import org.eclipse.fordiac.ide.model.libraryElement.INamedElement;
+import org.eclipse.fordiac.ide.model.libraryElement.SubApp;
+import org.eclipse.fordiac.ide.model.libraryElement.TypedSubApp;
+import org.eclipse.fordiac.ide.model.libraryElement.UntypedSubApp;
+import org.eclipse.fordiac.ide.model.ui.editors.HandlerHelper;
+import org.eclipse.fordiac.ide.ui.imageprovider.FordiacImage;
+import org.eclipse.gef.GraphicalEditPart;
+import org.eclipse.gef.GraphicalViewer;
+import org.eclipse.gef.commands.CompoundCommand;
+import org.eclipse.jface.viewers.DelegatingStyledCellLabelProvider.IStyledLabelProvider;
+import org.eclipse.jface.viewers.ISelection;
+import org.eclipse.jface.viewers.LabelProvider;
+import org.eclipse.jface.viewers.StyledString;
+import org.eclipse.jface.viewers.TreeNode;
+import org.eclipse.swt.graphics.Image;
+import org.eclipse.swt.graphics.Point;
+import org.eclipse.ui.IEditorPart;
+import org.eclipse.ui.ISources;
+import org.eclipse.ui.handlers.HandlerUtil;
+
+public class MoveThroughHierarchyHandler extends AbstractHandler {
+	@Override
+	public Object execute(final ExecutionEvent event) throws ExecutionException {
+		final IEditorPart editor = HandlerUtil.getActiveEditor(event);
+
+		if (null != editor) {
+			final ISelection selection = HandlerUtil.getCurrentSelection(event);
+			final List<FBNetworkElement> fbelements = HandlerHelper.getSelectedFBNElements(selection);
+
+			if (!fbelements.isEmpty()) {
+				// get selected FBNetwork
+				final SubAppHierarchyDialog dialog = new SubAppHierarchyDialog(fbelements.get(0));
+				final FBNetwork destinationNetwork = dialog.open();
+				if (destinationNetwork == null) {
+					return Status.CANCEL_STATUS;
+				}
+
+				final Point destination = getDestination(editor, fbelements, destinationNetwork);
+				final MoveElementsFromSubAppCommand cmd = new MoveElementsFromSubAppCommand(fbelements, destination,
```

Comment:
Maybe it would be an Idea to wrap all the collected commands into an 
MoveElemenThroughHierachy Command.
That will probably also make UNDO/REDO easier to implement

Reply:
I have not looked into undo/redo yet.
This should be easily fixed when this is merged with the Command for MoveToParent

Reply:
Should work now